### PR TITLE
[BugFix] fix incorrect restore of CPPFLAGS

### PR DESCRIPTION
**Pull request by kevincai**
* get rid of anonying warning messages 
 ```
 cc1: warning: command-line option '-Wno-class-memaccess' is valid for C++/ObjC++ but not for C
  ```
* bug introduced in pr #4517

## What type of PR is this:
- [X] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes:

Fixes #

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [X] I have checked the version labels which the pr will be auto backported to target branch
  - [X] 3.0
  - [x] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

File: thirdparty/build-thirdparty.sh [hunk 1]
```diff
@@ -187,15 +180,18 @@ build_openssl() {
...
+    export CFLAGS="-O3 -fno-omit-frame-pointer -fPIC"
+
     LDFLAGS="-L${TP_LIB_DIR}" \
     LIBDIR="lib" \
     ./Configure --prefix=$TP_INSTALL_DIR -lz -no-shared ${OPENSSL_PLATFORM}
     make -j$PARALLEL
     make install_sw
 
-    export CXXFLAGS=$OLD_FLAGS
-    export CPPFLAGS=$OLD_FLAGS
```
Comment (kevincai) [inline thirdparty/build-thirdparty.sh:197]: original code wrongly restore `CPPFLAGS` to `CXXFLAGS`

**Review by imay — APPROVED** [review]

File: thirdparty/build-thirdparty.sh [hunk 2]
```diff
@@ -788,12 +771,9 @@ build_breakpad() {
     cd $TP_SOURCE_DIR/$BREAK_PAD_SOURCE
     mkdir -p src/third_party/lss
     cp $TP_PATCH_DIR/linux_syscall_support.h src/third_party/lss
-    OLD_FLAGS=$CFLAGS
-    unset CFLAGS
-    ./configure --prefix=$TP_INSTALL_DIR --enable-shared=no --disable-samples --disable-libevent-regress
+    CFLAGS= ./configure --prefix=$TP_INSTALL_DIR --enable-shared=no --disable-samples --disable-libevent-regress
     make -j$PARALLEL
     make install
```
Comment (chaoyli) [inline thirdparty/build-thirdparty.sh:776]: Why not need to reset `CFLAGS` here?

File: thirdparty/build-thirdparty.sh [hunk 3]
```diff
@@ -371,19 +367,15 @@ build_gperftools() {
...
-
     if [ ! -f configure ]; then
         ./autogen.sh
     fi
 
     LDFLAGS="-L${TP_LIB_DIR}" \
+    CFLAGS="-O3 -fno-omit-frame-pointer -fPIC -g" \
     ./configure --prefix=$TP_INSTALL_DIR/gperftools --disable-shared --enable-static --disable-libunwind --with-pic --enable-frame-pointers
     make -j$PARALLEL
     make install
```
Comment (chaoyli) [inline thirdparty/build-thirdparty.sh:378]: Why not need to reset `CFLAGS` here?

Comment (kevincai) [reply thirdparty/build-thirdparty.sh:378]: can either unset and export a new flags or use one line format to temporarily change the CFLAGS vaule.

**Comment by kevincai** [comment]
@mergifyio rebase

**Comment by kevincai** [comment]
code rebased

**Review by imay — APPROVED** [review]

**Review by chaoyli — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Pull request merged**